frontend/components/ActivityItem/ActivityItem.tsx

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import ReactTooltip from "react-tooltip";
+import classnames from "classnames";
+
+import { IActivity, IActivityDetails } from "interfaces/activity";
+import {
+  addGravatarUrlToResource,
+  internationalTimeFormat,
+} from "utilities/helpers";
+import { DEFAULT_GRAVATAR_LINK } from "utilities/constants";
+
+import Avatar from "components/Avatar";
+
+import { COLORS } from "styles/var/colors";
+import { dateAgo } from "utilities/date_format";
+import Button from "components/buttons/Button";
+import Icon from "components/Icon";
+import { noop } from "lodash";
+
+const baseClass = "activity-item";
+
+export interface IShowActivityDetailsData {
+  type: string;
+  details?: IActivityDetails;
+}
+
+/**
+ * A handler that will show the details of an activity. This is used to pass
+ * the details of an activity to the parent component to show the details of
+ * the activity.
+ */
+export type ShowActivityDetailsHandler = ({
+  type,
+  details,
+}: IShowActivityDetailsData) => void;
+
+interface IActivityItemProps {
+  activity: IActivity;
+  children: React.ReactNode;
+  /**
+   * Set this to `true` when rendering only this activity by itself. This will
+   * change the styles for the activity item for solo rendering.
+   * @default false */
+  soloActivity?: boolean;
+  /**
+   * Set this to `true` to hide the show details button and prevent from rendering.
+   * Not all activities can show details, so this is a way to hide the button.
+   * @default false
+   */
+  hideShowDetails?: boolean;
+  /**
+   * Set this to `true` to hide the close button and prevent from rendering
+   * @default false
+   */
+  hideClose?: boolean;
```

Comment:
should we rename this to `hideCancel`? Because it's hiding the ability to cancel an activity and close is just the icon name, but it's not really closing the activity persay?

Reply:
yes, this was missnamed. will make that change.